specs/boot_loader_specification.md

```diff
@@ -279,6 +279,35 @@ The following keys are recognized:
 * `efi` refers to an arbitrary EFI program. If this key is set, and the system
   is not an EFI system, this entry should be hidden.
 
+* `uki` is a combination of the `linux` key, the `efi` key and Type #2 semantics: if this key is used the
+  specified path shall refer to a [unified kernel image (UKI)](unified_kernel_image.md). This informs the
+  boot loader to use UKI semantics (as exposed by Type #2) even though it is referenced via a Type #1
+  entry. This is useful to parameterize the UKI differently, to place it in a directory other than
+  `/EFI/Linux`, or to share a single UKI image among multiple menu entries.
+
+  If this key is set, and the system is not an EFI system, this entry should be hidden.
+
+  Example: `uki /fooos/bar.efi`
+
+* `uki-url` is similar to `uki`, but takes an RFC 3986 URI instead of a file path. It may be used to
```

Comment:
IPXE supports specifiers in URLs e.g.

```
#!ipxe
initrd http://myserver.com/initrd?uuid=${uuid}&mac=${mac:hexhyp}
chain https://myserver.com/vmlinuz?uuid=${uuid}&mac=${mac:hexhyp}
```

which is great in datacenters as you can server specific images for specific servers dynamically.
e.g. with https://matchbox.psdn.io/matchbox/

Can we do something similar here?

Reply:
So a specifier for the machine-id and maybe the SMBIOS UUID (Though perhaps those are the same when SMBIOS UUID is available?)

Reply:
Fits in nicely with: https://github.com/systemd/systemd/pull/32086 because then the machine id and the SMBIOS UUID match